scm_v3c/bootload/bootload.py

```diff
@@ -78,8 +212,39 @@ def program_cortex(teensy_port="COM15", scum_port="COM18", binary_image="./code.
     # Transfer payload to Teensy
     teensy_ser.write(b'transfersram\n')
     print(teensy_ser.readline())
-    # Send the binary data over uart
-    teensy_ser.write(bindata)
+    teensy_ser.write(b'\r\n')
+    print(teensy_ser.readline())
+    time.sleep(0.1)
+    # We seem to have a limit of 64 total packet length
+    chunk_size = 16  # Each byte is represented by 2 hex characters
+                     # chunk_size = 16 results in a packet of 41
+    for i in range(0, len(bindata), chunk_size):
+        chunk = bindata[i:i+chunk_size]
+        send_bin_packet(teensy_ser, chunk)
+
+    # Wait for the final confirmation
+    resp = teensy_ser.readline()
+    while(len(resp) == 0):
```

Comment:
C >> Python

Reply:
where do while 😭

Reply:
Oh I just meant the braces aren't needed

Reply:
ah gotcha. do while would also make that way more clean.

Reply:
Revised